server/controllers/logger.ts

```diff
@@ -22,51 +22,55 @@ interface UserData {
   recommendation: string;
 }
 
-//func to insert and get data from the database
-export const insertUserData = async (
-  userData: UserData,
-): Promise<UserData | null> => {
-  const { data, error } = await supabase.from('user_data').insert([userData]);
+export const insertUserDataMiddleware = async (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+) => {
+  const userData: UserData = res.locals;
 
-  if (error) {
-    console.error('Error inserting data:', error);
-    return null;
+  if (!userData) {
+    return next(error);
   }
-  console.log('Data inserted successfully:', data);
-  return data[0];
-};
 
-// Fetch all user data from the database
-export const getUserData = async (): Promise<UserData[] | null> => {
-  const { data, error } = await supabase.from('user_data').select('*');
+  try {
+    const { data, error } = await supabase.from('user_data').insert([userData]);
+
+    if (error) {
+      console.error('Error inserting data:', error);
+      return res.status(500).json({ error: 'Failed to insert data' });
+    }
 
-  if (error) {
-    console.error('Error retrieving data:', error);
-    return null;
+    console.log('Data inserted successfully:', data);
+    // Store inserted data in `res.locals` for further middleware
+    res.locals.insertedData = data[0];
```

Comment:
Because we are pulling all the data off `res.locals`, I don't think we need to put the same data back on it.